ompi/mca/coll/base/coll_base_util.c

```diff
@@ -297,11 +298,232 @@ int ompi_coll_base_retain_datatypes_w( ompi_request_t *req,
+static void nbc_req_cons(ompi_coll_base_nbc_request_t *req)
+{
     req->cb.req_complete_cb = NULL;
     req->req_complete_cb_data = NULL;
     req->data.objs.objs[0] = NULL;
     req->data.objs.objs[1] = NULL;
 }
 
 OBJ_CLASS_INSTANCE(ompi_coll_base_nbc_request_t, ompi_request_t, nbc_req_cons, NULL);
+
+/* File reading functions */
+static void skiptonewline (FILE *fptr, int *fileline)
+{
+    char val;
+    int rc;
+
+    do {
+        rc = fread(&val, 1, 1, fptr);
+        if (0 == rc) {
+            return;
+        }
+        if ('\n' == val) {
+            (*fileline)++;
+            return;
+        }
+    } while (1);
+}
+
+int ompi_coll_base_file_getnext_long(FILE *fptr, int *fileline, long* val)
+{
+    char trash;
+    int rc;
+
+    do {
+        rc = fscanf(fptr, "%li", val);
+        if (rc == EOF) {
+            return -1;
+        }
+        if (1 == rc) {
+            return 0;
+        }
+        /* in all other cases, skip to the end of the token */
+        rc = fread(&trash, sizeof(char), 1, fptr);
+        if (rc == EOF) {
+            return -1;
+        }
+        if ('\n' == trash) (*fileline)++;
+        if ('#' == trash) {
+            skiptonewline (fptr, fileline);
+        }
+    } while (1);
+}
+
+int ompi_coll_base_file_getnext_string(FILE *fptr, int *fileline, char** val)
+{
+    char trash, token[32];
+    int rc;
+
+    *val = NULL;  /* security in case we fail */
+    do {
+        rc = fscanf(fptr, "%32s", token);
+        if (rc == EOF) {
+            return -1;
+        }
+        if (1 == rc) {
+            if( '#' == token[0] ) {
+                skiptonewline(fptr, fileline);
+                continue;
+            }
+            *val = (char*)malloc(strlen(token) + 1);
+            strcpy(*val, token);
+            return 0;
+        }
+        /* in all other cases, skip to the end of the token */
+        rc = fread(&trash, sizeof(char), 1, fptr);
+        if (rc == EOF) {
+            return -1;
+        }
+        if ('\n' == trash) (*fileline)++;
+        if ('#' == trash) {
+            skiptonewline (fptr, fileline);
+        }
+    } while (1);
+}
+
+int ompi_coll_base_file_getnext_size_t(FILE *fptr, int *fileline, size_t* val)
+{
+    char trash;
+    int rc;
+
+    do {
+        rc = fscanf(fptr, "%" PRIsize_t, val);
+        if (rc == EOF) {
+            return -1;
+        }
+        if (1 == rc) {
+            return 0;
+        }
+        /* in all other cases, skip to the end of the token */
+        rc = fread(&trash, sizeof(char), 1, fptr);
+        if (rc == EOF) {
+            return -1;
+        }
+        if ('\n' == trash) (*fileline)++;
+        if ('#' == trash) {
+            skiptonewline (fptr, fileline);
+        }
+    } while (1);
+}
+
+int mca_coll_base_name_to_colltype(const char* name)
+{
+    if( 0 == strncmp(name, "neighbor_all", 12) ) {
+        if( 't' != name[12] ) {
+            if( 0 == strcmp(name+12, "allgatherv") )
```

Comment:
I assume this to be wrong and it should be `name+9` at this and the following paces

Reply:
good catch, the 12 is correct but the "all"  should have been removed from the strcmp.

Reply:
Seems fixed for me.